### ISSUE TITLE
USB issues on certain computers

### ISSUE THREAD

**Issue by francoisferland** [ohm-ros-pkg/optris_drivers#15]
Hi,

First, I apologize if this is not the right venue to discuss this, but Optris told me you were the maintainers for the Linux drivers.  My issue is not necessarily ROS-specific, but we have problem connecting our PI-640 camera on various recent laptops and Mini-ITX computers running Ubuntu 14.04 (64-bit).

As you suggest, we tried isolating the camera on its own bus and other uvc options, but we just cannot make it work. In some cases, the camera works correctly in Windows, but not Linux on the same laptop.

We did manage to locate two older laptops that seem to work with it in Ubuntu 14.04, but on our robots we would much prefer connecting the camera directly to their embedded computers instead of having to use a separate computer just for one sensor. 

Thanks in advance for your help!

**Comment by stefanmay** [comment]
Hi,

feel free to contact me directly. In order to help you, I need more 
information about your system and the specific problem. There are 
multiple issues that might be the reason for it. First of all, can you 
grab images with guvcview, i.e. do you see the typical green raw image 
in a high frame rate? If not, this issue is related to the uvcvideo 
module. If yes, I need more insight in your system (installed 
calibration files, xml configration, file permissions, user permissions, 
...)

**Comment by francoisferland** [comment]
Hi Stefan,

Yes, the problem is related to uvcvideo, i.e., guvcview cannot grab anything (high frame rate or not). On PCs that work correctly everything else is fine. The only thing we noticed about those working PCs is they tend to be slightly older, I would say roughly 3 years old. Anything that has a USB 3.0 port usually does not work (even when the camera is plugged on a USB 2.0-only bus).

**Comment by stefanmay** [comment]
Dear François,

this seems to be a general problem. Can you stream a different UVC 
camera, e.g. a webcam via these ports?
Is there a BIOS option, that might solve the problem (something related 
to USB 3.0)?

Best regards,

    Stefan

**Comment by francoisferland** [comment]
Dear Stefan,

Yes, other cameras usually work without issues, some with higher resolutions than the Optris. I will look into the BIOS and report back. I will also try to provide you with logs of what it looks like when we try to stream the Optris.

Best regards,

François

**Comment by francoisferland** [comment]
Dear Stefan,

Something strange happened, but somehow I am no longer able to reproduce the problem. It's also working pretty well on USB 3.0 while sharing a bus with a Kinect, which usually takes a good amount of bandwidth. I didn't look in the BIOS settings, but I did update my Ubuntu installation (still on 14.04) in the meantime. Maybe something was patched in uvcvideo in the past weeks/months? Still, it's a strange coincidence. To be sure, I tested it on another computer that I remember had issues with the camera a few months ago, and the problem is also gone there, so I guess it's a good thing!

However, we have colleagues to whom we lent our camera a few weeks ago and who did experience similar issues in Linux. They are in the process of ordering their own Optris imager, but I'm not sure if they received it yet. In any case, I will point them here to see if the issue is also gone on their side.

Thanks again!

Best regards,

François
